modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1174,8 +1174,8 @@ def astype_builder(df):
             new_frame,
             self.index,
             self.columns,
```

Comment:
I wonder if we should pass `self._index_cache` and `self._columns_cache` here, too

Reply:
Good catch! Will update